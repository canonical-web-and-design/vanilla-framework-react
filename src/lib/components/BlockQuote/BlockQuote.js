import React from 'react';
import PropTypes from 'prop-types';

const BlockQuote = (props) => {
  if (props.pull) {
    return (
      <blockquote className="p-pull-quote">
        <p>{props.quote}</p>
        <cite className="p-pull-quote__citation">{props.citation}</cite>
      </blockquote>
    );
  }

  return (
    <blockquote>
      <p>
        <span>&quot;</span>{props.quote}<span>&quot;</span>
      </p>
      <p>
        <cite>{props.citation}</cite>
      </p>
    </blockquote>
  );
};

BlockQuote.defaultProps = {
  citation: '',
  pull: false,
};

BlockQuote.propTypes = {
  quote: PropTypes.string.isRequired,
  citation: PropTypes.string,
  pull: PropTypes.bool,
};

BlockQuote.displayName = 'BlockQuote';

export default BlockQuote;
