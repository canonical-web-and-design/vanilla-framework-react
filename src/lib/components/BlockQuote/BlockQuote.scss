@import 'vanilla-framework/scss/base';
@import 'vanilla-framework/scss/patterns_pull-quotes';
@include vf-b-typography;
@include vf-b-blockquotes;
@include vf-p-pull-quotes;
