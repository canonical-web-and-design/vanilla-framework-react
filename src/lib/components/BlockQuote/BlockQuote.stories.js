import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import BlockQuote from './BlockQuote';

storiesOf('Block Quote', module)
  .add('Default',
    withInfo('Basic blockquotes and citations.')(() => (
      <BlockQuote
        quote="Ubuntu is an ancient African word meaning 'humanity to others'.
        Ubuntu is an ancient African word meaning 'humanity to others'."
        citation="Canonical"
      />),
    ),
  )

  .add('Pull Quote',
    withInfo('Use the pull quote pattern to highlight content from different sources in a visually prominent way.')(() => (
      <BlockQuote
        pull
        quote="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor
        incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam"
        citation="A. N. Author"
      />),
    ),
  );
