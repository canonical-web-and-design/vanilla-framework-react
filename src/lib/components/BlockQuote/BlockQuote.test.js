import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import BlockQuote from './BlockQuote';

describe('BlockQuote component', () => {
  it('should render a basic blockquote correctly', () => {
    const blockQuote = ReactTestRenderer.create(
      <BlockQuote
        quote="Ubuntu is an ancient African word meaning 'humanity to others'.
        Ubuntu is an ancient African word meaning 'humanity to others'."
      />,
    );
    const json = blockQuote.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a basic blockquote with citation correctly', () => {
    const blockQuote = ReactTestRenderer.create(
      <BlockQuote
        quote="Ubuntu is an ancient African word meaning 'humanity to others'.
        Ubuntu is an ancient African word meaning 'humanity to others'."
        citation="Canonical"
      />,
    );
    const json = blockQuote.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a pull quote correctly', () => {
    const blockQuote = ReactTestRenderer.create(
      <BlockQuote
        pull
        quote="Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor
        incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam"
        citation="A. N. Author"
      />,
    );
    const json = blockQuote.toJSON();
    expect(json).toMatchSnapshot();
  });
});
