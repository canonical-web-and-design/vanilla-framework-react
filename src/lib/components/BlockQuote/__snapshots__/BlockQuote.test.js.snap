// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockQuote component should render a basic blockquote correctly 1`] = `
<blockquote>
  <p>
    <span>
      "
    </span>
    Ubuntu is an ancient African word meaning 'humanity to others'. Ubuntu is an ancient African word meaning 'humanity to others'.
    <span>
      "
    </span>
  </p>
  <p>
    <cite>
      
    </cite>
  </p>
</blockquote>
`;

exports[`BlockQuote component should render a basic blockquote with citation correctly 1`] = `
<blockquote>
  <p>
    <span>
      "
    </span>
    Ubuntu is an ancient African word meaning 'humanity to others'. Ubuntu is an ancient African word meaning 'humanity to others'.
    <span>
      "
    </span>
  </p>
  <p>
    <cite>
      Canonical
    </cite>
  </p>
</blockquote>
`;

exports[`BlockQuote component should render a pull quote correctly 1`] = `
<blockquote
  className="p-pull-quote"
>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam
  </p>
  <cite
    className="p-pull-quote__citation"
  >
    A. N. Author
  </cite>
</blockquote>
`;
